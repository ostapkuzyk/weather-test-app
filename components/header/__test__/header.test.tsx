import Header from '../index';
import { render } from '@testing-library/react-native';

jest.mock('@/context/AppContext', () => ({
  useAppContext: jest.fn().mockReturnValue({
    address: {
      name: 'name',
      lat: 0,
      lon: 0
    }
  })
}));

describe('header', () => {
  it('renders without crashing', () => {
    const comp = render(<Header />);
    expect(comp.toJSON()).toMatchSnapshot();
  });
});