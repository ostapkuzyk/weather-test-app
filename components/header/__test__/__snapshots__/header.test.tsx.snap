// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders without crashing 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
  testID="header"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="40"
      bbWidth="40"
      fill="white"
      focusable={false}
      height="40"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 40,
            "width": 40,
          },
        ]
      }
      vbHeight={1024}
      vbWidth={1024}
      width="40"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M512 911.529l-234.041-234.041c-46.289-46.289-77.811-105.263-90.581-169.469-12.77-64.203-6.215-130.752 18.835-191.231s67.474-112.171 121.903-148.539c54.43-36.368 118.422-55.779 183.882-55.779s129.454 19.411 183.881 55.779c54.43 36.368 96.853 88.061 121.905 148.539 25.051 60.479 31.605 127.028 18.835 191.231-12.772 64.206-44.294 123.18-90.583 169.469l-234.041 234.041zM694.042 625.489c35.998-36.004 60.514-81.873 70.443-131.811 9.932-49.936 4.831-101.694-14.651-148.731-19.486-47.038-52.48-87.241-94.816-115.526-42.332-28.285-92.102-43.382-143.016-43.382s-100.684 15.098-143.017 43.382c-42.333 28.285-75.33 68.488-94.814 115.526s-24.585 98.795-14.653 148.731c9.932 49.936 34.446 95.806 70.446 131.811l182.040 182.038 182.042-182.038zM512 517c-19.506 0-38.213-7.747-52.008-21.543-13.791-13.795-21.543-32.5-21.543-52.008 0-19.506 7.75-38.213 21.543-52.008 13.795-13.791 32.504-21.543 52.008-21.543s38.213 7.75 52.008 21.543c13.791 13.795 21.543 32.504 21.543 52.008 0 19.508-7.75 38.213-21.543 52.008-13.795 13.795-32.504 21.543-52.008 21.543z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeWidth={5}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "color": "white",
          "flexDirection": "row",
          "fontSize": 26,
          "marginLeft": 12,
        }
      }
    >
      name
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="40"
      bbWidth="40"
      fill="white"
      focusable={false}
      height="40"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 40,
            "width": 40,
          },
        ]
      }
      vbHeight={1024}
      vbWidth={1024}
      width="40"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M511.998 374.325c-76.035 0-137.674 61.638-137.674 137.674s61.639 137.674 137.674 137.674c76.037 0 137.674-61.637 137.674-137.674s-61.637-137.674-137.674-137.674zM429.394 511.999c0-45.62 36.984-82.604 82.604-82.604s82.604 36.984 82.604 82.604c0 45.62-36.984 82.604-82.604 82.604s-82.604-36.984-82.604-82.604z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeWidth={5}
        />
        <RNSVGPath
          d="M511.069 117.332c-16.315 0-29.939 0-41.123 0.762-11.642 0.795-22.638 2.507-33.298 6.923-24.738 10.247-44.394 29.902-54.64 54.64-5.334 12.879-6.778 26.451-7.332 41.205-0.443 11.859-6.429 21.754-15.262 26.855s-20.393 5.334-30.887-0.21c-13.053-6.899-25.529-12.435-39.35-14.253-26.548-3.496-53.397 3.699-74.639 19.999-9.155 7.024-16.137 15.692-22.645 25.376-6.253 9.304-13.064 21.101-21.221 35.232l-0.927 1.606c-8.159 14.131-14.97 25.928-19.9 35.995-5.133 10.479-9.148 20.859-10.654 32.299-3.496 26.548 3.699 53.397 19.999 74.641 8.485 11.055 19.516 19.095 32.014 26.947 10.051 6.315 15.63 16.447 15.629 26.65 0 10.199-5.579 20.332-15.629 26.646-12.5 7.852-23.532 15.892-32.018 26.952-16.3 21.243-23.495 48.091-19.999 74.637 1.507 11.439 5.521 21.823 10.654 32.3 4.931 10.067 11.742 21.862 19.9 35.994l0.928 1.609c8.159 14.13 14.969 25.927 21.221 35.23 6.508 9.685 13.489 18.353 22.645 25.376 21.244 16.3 48.093 23.497 74.639 20.001 13.82-1.82 26.297-7.354 39.349-14.251 10.493-5.547 22.058-5.312 30.89-0.209 8.834 5.1 14.819 14.993 15.263 26.856 0.551 14.751 1.997 28.324 7.332 41.203 10.247 24.737 29.902 44.394 54.64 54.641 10.662 4.417 21.657 6.127 33.298 6.921 11.182 0.763 24.808 0.763 41.123 0.763h1.855c16.319 0 29.939 0 41.126-0.763 11.642-0.794 22.638-2.504 33.299-6.921 24.737-10.246 44.394-29.902 54.641-54.641 5.334-12.879 6.777-26.451 7.328-41.207 0.444-11.859 6.429-21.752 15.263-26.856 8.834-5.1 20.398-5.334 30.89 0.209 13.052 6.899 25.526 12.435 39.35 14.251 26.548 3.496 53.396-3.697 74.637-19.999 9.157-7.026 16.136-15.692 22.645-25.376 6.252-9.304 13.063-21.099 21.22-35.23l0.929-1.609c8.159-14.13 14.971-25.931 19.902-35.997 5.133-10.477 9.145-20.857 10.654-32.296 3.496-26.551-3.701-53.399-20.001-74.642-8.484-11.057-19.518-19.095-32.013-26.947-10.052-6.315-15.629-16.447-15.629-26.65 0-10.199 5.577-20.329 15.625-26.643 12.504-7.852 23.538-15.89 32.022-26.952 16.3-21.243 23.497-48.091 20.001-74.638-1.51-11.44-5.521-21.821-10.654-32.299-4.93-10.066-11.741-21.861-19.902-35.992l-0.925-1.608c-8.161-14.132-14.971-25.931-21.224-35.234-6.509-9.685-13.488-18.352-22.645-25.376-21.243-16.3-48.091-23.495-74.642-19.999-13.82 1.819-26.294 7.355-39.346 14.252-10.492 5.546-22.058 5.31-30.89 0.21s-14.821-14.996-15.265-26.856c-0.55-14.753-1.993-28.324-7.328-41.202-10.246-24.738-29.902-44.394-54.641-54.64-10.662-4.416-21.657-6.128-33.299-6.923-11.187-0.763-24.808-0.763-41.126-0.762h-1.855zM457.722 175.896c2.834-1.174 7.136-2.255 15.974-2.858 9.086-0.62 20.82-0.635 38.302-0.635s29.216 0.015 38.3 0.635c8.84 0.603 13.143 1.684 15.974 2.858 11.245 4.657 20.182 13.592 24.836 24.836 1.473 3.546 2.687 9.076 3.179 22.186 1.087 29.090 16.102 57.1 42.76 72.49 26.654 15.39 58.421 14.388 84.157 0.785 11.598-6.13 16.999-7.84 20.801-8.342 12.068-1.588 24.271 1.682 33.926 9.091 2.435 1.867 5.521 5.051 10.464 12.404 5.081 7.558 10.958 17.712 19.701 32.852s14.597 25.31 18.603 33.488c3.895 7.955 5.111 12.223 5.51 15.263 1.589 12.068-1.682 24.271-9.090 33.927-2.338 3.046-6.517 6.865-17.631 13.844-24.646 15.489-41.395 42.495-41.397 73.272 0 30.78 16.749 57.79 41.397 73.279 11.109 6.979 15.287 10.798 17.626 13.844 7.408 9.655 10.68 21.86 9.090 33.926-0.401 3.040-1.615 7.307-5.51 15.263-4.005 8.176-9.861 18.346-18.603 33.486s-14.62 25.295-19.701 32.854c-4.941 7.35-8.029 10.537-10.464 12.401-9.655 7.412-21.86 10.68-33.926 9.095-3.804-0.503-9.205-2.214-20.801-8.342-25.735-13.606-57.504-14.605-84.161 0.785-26.65 15.39-41.666 43.398-42.753 72.489-0.492 13.11-1.708 18.64-3.179 22.19-4.656 11.242-13.591 20.178-24.836 24.836-2.831 1.171-7.134 2.254-15.974 2.855-9.084 0.621-20.816 0.635-38.3 0.635s-29.216-0.015-38.302-0.635c-8.837-0.602-13.14-1.684-15.974-2.855-11.242-4.658-20.177-13.595-24.834-24.836-1.47-3.55-2.687-9.079-3.177-22.19-1.088-29.088-16.102-57.099-42.759-72.489s-58.421-14.387-84.158-0.785c-11.6 6.131-16.997 7.842-20.802 8.342-12.068 1.589-24.271-1.682-33.927-9.090-2.433-1.865-5.521-5.051-10.463-12.405-5.079-7.555-10.958-17.71-19.701-32.85s-14.596-25.31-18.601-33.486c-3.897-7.956-5.112-12.226-5.511-15.265-1.588-12.068 1.682-24.271 9.091-33.926 2.337-3.043 6.518-6.865 17.626-13.844 24.648-15.485 41.398-42.495 41.399-73.272 0.002-30.784-16.749-57.794-41.399-73.283-11.106-6.979-15.287-10.798-17.623-13.844-7.409-9.656-10.679-21.861-9.091-33.927 0.401-3.040 1.614-7.308 5.511-15.263 4.005-8.178 9.86-18.346 18.601-33.488s14.621-25.294 19.701-32.852c4.941-7.352 8.030-10.539 10.463-12.404 9.656-7.409 21.861-10.679 33.927-9.091 3.806 0.5 9.205 2.212 20.805 8.344 25.735 13.602 57.5 14.604 84.155-0.785s41.67-43.397 42.758-72.488c0.49-13.112 1.708-18.642 3.177-22.188 4.657-11.244 13.594-20.179 24.834-24.836z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeWidth={5}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
